test/challtestsrv/dnsone.go

```diff
@@ -0,0 +1,171 @@
+package challtestsrv
+
+import (
+	"net"
+	"os"
+	"time"
+
+	"github.com/miekg/dns"
+)
+
+// AddDNSOneChallenge adds a TXT record for the given host with the given
+// content.
+func (s *ChallSrv) AddDNSOneChallenge(host, content string) {
+	s.dnsMu.Lock()
+	defer s.dnsMu.Unlock()
+	s.dnsOne[host] = append(s.dnsOne[host], content)
+}
+
+// DeleteDNSOneChallenge deletes a TXT record for the given host.
+func (s *ChallSrv) DeleteDNSOneChallenge(host string) {
+	s.dnsMu.Lock()
+	defer s.dnsMu.Unlock()
+	if _, ok := s.dnsOne[host]; ok {
+		delete(s.dnsOne, host)
+	}
+}
+
+// GetDNSOneChallenge returns a slice of TXT record valuefor the given host and
+// a true bool. If the host does not exist in the challenge response data then
```

Comment:
It seems like it would be reasonable to just have a single return value: a slice of strings, which would be empty (nil) if there are no corresponding records.